### ISSUE TITLE
import issue when wrapping outside dependencies

### ISSUE THREAD

**Issue by tehsphinx** [99designs/gqlgen#649]
### General Setup

I have two different `Status` structs in different packages with overlapping properties and want to show them as an `interface`. `gqlgen` generates the `models` interface:

```
type Status interface {
	IsStatus()
}
```

So far everything is perfect.

Now those two structs don't implement `Status` as they are missing `IsStatus` method. Since I did not want to change the packages where those status are located I thought I'd be clever and create a small wrapper model:

```
// DecoderStatus is a lightweight wrapper for decoder status.
type DecoderStatus struct {
	decstat.StatusData
}

// IsStatus needed to implement Status.
func (s *DecoderStatus) IsStatus() {}
```

And now the problems start...

### Expected Behaviour

`gqlgen generate` runs without any issues.

### Actual Behavior

`gqlgen generate` panics:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x1308fa9]

goroutine 1 [running]:
github.com/99designs/gqlgen/codegen/config.(*Binder).ObjectPosition(0xc001154640, 0x16c8ca0, 0xc000cea820, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/gn/repos/gqlgen/codegen/config/binder.go:68 +0x209
github.com/99designs/gqlgen/codegen.(*builder).bindField(0xc00022d698, 0xc000be0580, 0xc000f3e240, 0x0, 0x0)
	/Users/gn/repos/gqlgen/codegen/field.go:116 +0x3b8
github.com/99designs/gqlgen/codegen.(*builder).buildField(0xc00022d698, 0xc000be0580, 0xc0008c41c0, 0x0, 0x1, 0xc000a2d6d8)
	/Users/gn/repos/gqlgen/codegen/field.go:64 +0x31d
github.com/99designs/gqlgen/codegen.(*builder).buildObject(0xc00022d698, 0xc00025f380, 0x15d93b3, 0x6, 0x0)
	/Users/gn/repos/gqlgen/codegen/object.go:73 +0x4d4
github.com/99designs/gqlgen/codegen.BuildData(0xc0000cc240, 0x16b45c0, 0x1a9f808, 0x0)
	/Users/gn/repos/gqlgen/codegen/data.go:84 +0x461
github.com/99designs/gqlgen/api.Generate(0xc0000cc240, 0x0, 0x0, 0x0, 0x0, 0xc0000b8281)
	/Users/gn/repos/gqlgen/api/generate.go:37 +0x22d
github.com/99designs/gqlgen/cmd.glob..func1(0xc0000e42c0)
	/Users/gn/repos/gqlgen/cmd/gen.go:39 +0xa2
github.com/urfave/cli.HandleAction(0x151cb20, 0x1618f58, 0xc0000e42c0, 0xc0000c6a00, 0x0)
	/Users/gn/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:492 +0x7c
github.com/urfave/cli.Command.Run(0x15de05e, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1609eb5, 0x29, 0x0, ...)
	/Users/gn/go/pkg/mod/github.com/urfave/cli@v1.20.0/command.go:210 +0x92f
github.com/urfave/cli.(*App).Run(0xc0000bc680, 0xc0000a8180, 0x3, 0x3, 0x0, 0x0)
	/Users/gn/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:255 +0x69b
github.com/99designs/gqlgen/cmd.Execute()
	/Users/gn/repos/gqlgen/cmd/root.go:40 +0x224
main.main()
	/Users/gn/repos/gqlgen/main.go:8 +0x20
```

Digging deeper and debugging what is happening it turns out that the `decstat` pkg is missing since it is nowhere mentioned in the config... but it is imported by the `DecoderStatus` model.

If I add an extra model to the config file that is not needed anywhere it works:

```
models:
  DecStat:
    model: example.com/lib/timing/decstat.StatusData
```

Is this expected and the way to solve this or should `gqlgen` also import packages imported by the models?

**Comment by vektah** [comment]
So the interface is generated, but the implementations are in a package you dont want to modify?

First let me say this is definitly a bug, and we should get it fixed.

Did you consider a name-only empty interface? Might fit better if you dont want to change the underlying code?
```go
type Status interface {}
```
and in config
```yaml
models:
  Status: {model: "example.com/your/app.Status"}
```

**Comment by tehsphinx** [comment]
A name only interface would probably work, yes. But the interface is still generated every time:

```
type Status interface {
	IsStatus()
}
```

But since I found my workaround (with the extra, unused models in the config file) I am happy for now. Just thought it might be good to fix this. 😏

**Comment by wildlifehexagon** [comment]
I had a similar issue when pointing my `gqlgen.yml` file to external models. In my GraphQL schema I had included an `interface` which `gqlgen` generated a Go interface for. The problem was that this interface was expecting two methods which did not exist in my external code. Since that external code was generated by `go-genproto`, it wasn't something I could just add in manually.

My workaround was basically the same as @tehsphinx -- I created a light wrapper for my external model and pointed `gqlgen.yaml` to that instead, then updated my resolver code accordingly.

Happy to provide more details if desired.
